examples/download_and_move_file.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'uber_task'
+require 'colorize'
+
+require_relative 'download_file'
+require_relative 'move_file'
+
+module Examples
+  class DownloadAndMoveFile
+    def self.run(source:, download_path:, move_path:)
+      UberTask.run('Download and move file', retry_count: 2) do
+        UberTask.on_success do
+          UberTask.logger.info(
+            'Top-level task was completed successfully!'.green,
+          )
+        end
+
+        UberTask.on_subtask_error do
+          UberTask.logger.info(
+            'Unexpected subtask error can be caught on top-level task ' \
+            'and an early return can be made, ' \
+            'isn\'t that cool!?'.underline,
+          )
+          return # Try to remove this line and see what happens!
```

Comment:
Remove the return, returns inside this block will affect the method that handles the [exection](https://github.com/shakacode/uber_task/blob/main/lib/uber_task/task_context.rb#L73).

We have to raise [RetryTask](https://github.com/shakacode/uber_task/blob/main/lib/uber_task.rb#L76), [SkipTask ](https://github.com/shakacode/uber_task/blob/main/lib/uber_task.rb#L126)or [EventHandled](https://github.com/shakacode/uber_task/blob/main/lib/uber_task/task_context.rb#L121C14-L121C26). 

It looks like I forgot to add a UberTask.event_handled method, add that method.

Reply:
Removed the return and added handling different types of exceptions which can be raised by subtasks